### PR TITLE
Docs update: simplify graph types narrative and make all the references consistent

### PR DESCRIPTION
**Pull request by max-causalens**
## Description and Motivation

Structure the DAG/CPDAG/MAG/PAG discussion in a consistent set of paragraphs and a summary table.

## Additional Context

Small improvement to readability / understandability / lower the academicness of the documentation.

## How Has This Been Tested?

Has not been tested at all.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which improves functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactor (moving code around, general refactoring improvements)
- [X] Documentation (documentation)

## PR Checklist:

- [x] I have reviewed my own PR? (review the PR as you are reviewing someone else's PR)
- [ ] I have implemented all requirements? (see JIRA, project documentation)
- [ ] I am affecting someone else's work? If so: I have added those people as reviewers?
- [ ] I have added comments to all the bits that are hard to follow
- [ ] At a bare minimum, I tested this manually
- [ ] I have added unit test(s)
- [ ] Is this a bugfix? If so have I added a regression test?
- [x] Does this PR satisfy the [one PR, one concern](https://medium.com/@fagnerbrack/one-pull-request-one-concern-e84a27dfe9f1) rule?
- [ ] If needed, documentation has been added/updated?
- [ ] I have updated the appropriate changelog with a line for my changes
- [ ] If this PR breaks reproducibility, have I added a note in the changelog explaining the break in reproducibility
      and its extent?

### CODE REVIEW

**Review by maxelliott-causalens — APPROVED** [review]
Very useful!

**Review by ilya-causalens — APPROVED** [review]

File: docs/introduction.md [hunk 1]
```diff
@@ -33,35 +33,48 @@ class.
...
+resolved to either `A -> B` or `A <- B`.
 
-One prominent representation of _MECs_ is a _Completed Partially Directed Acyclic Graph_ (_CPDAG_), which only contains
-directed (`->`) and undirected (`--`) edges. In this case, an undirected edge simply implies that a causal relationship
-can point either way, i.e., `A -- B` can be resolved to either `A -> B` or `A <- B`. In the above example, the _CPDAG_
-_MEC_ of the two _DAGs_ is therefore `A -- B -- C`.
+A _Maximal Ancestral Graph_ (_MAG_) can encode all the information that a _CPDAG_ can, but also provides
+information such as whether a latent confounder is likely to exist or selection bias is likely to be present. Specifically,
+_MAGs_ may also contain bi-directed edges `A <> B`, which imply the existence of a latent confounder between the respective
+variables.
```
Comment (andrew-causalens) [inline docs/introduction.md:46]: Should we say that `A -- B` implies selection bias?

File: docs/introduction.md [hunk 2]
```diff
@@ -33,35 +33,48 @@ class.
...
-more detailed information about the relationships between variables, such as including whether a latent confounder is
-likely to exist or selection bias is likely to be present. Specifically, _MAGs_ may also contain bi-directed edges
-`A <> B`, which implies that there is a latent confounder between the respective variables. A _Partial Ancestral Graph_ (_PAG_) 
-describes an equivalence class of _MAGs_. _PAGs_ may also contain "wild-card" or "circle" edges `A -o B`, which can 
-either be a directed or undirected arrow head, i.e. `A -o B` can be  resolved to `A -- B` or `A -> B`. The `o` end is r
-eferenced to as "unknown" in this package. See `cai_causal_graph.type_definitions.EdgeType` for all the supported edge 
-types. Similar to a _CPDAG_, a _PAG_ can represent  a number of _DAGs_. Therefore, _PAGs_, _CPDAGs_ and _DAGs_ can be 
-thought of in a hierarchical way.
+A _Partial Ancestral Graph_ (_PAG_) describes an equivalence class of _MAGs_. _PAGs_ may also contain "wild-card" or
+"circle" edges (`-o`), which can either be a directed or undirected arrow head, i.e. `A -o B` can be  resolved to
```
Comment (andrew-causalens) [inline docs/introduction.md:49]: "be  resolved" -> "be resolved"

File: docs/introduction.md [hunk 3]
```diff
@@ -33,35 +33,48 @@ class.
...
+Multiple _DAGs_/_CPDAGs_/_MAGs_/_PAGs_ can be consistent with the same _MEC_. For instance, if you identify the
+graphical structure `X -> Y -> Z`, then corresponding data would show that `X` **is** independent of `Z` given `Y`.
+However, the graphical structures `X <- Y <- Z` and `X <- Y -> Z` would lead to the exact same conditional independence
+test result as above. Only if the graphical structure found was a collider connection `X -> Y <- Z` would you be able to
+identify the structure from observational data, because the data would tell you that `X` and `Z` are independent, but
+become dependent given `Y`.
+:::
+
+:::warning
+In a `CPDAG` the `--` implies an existence of an edge which can be in either direction, `<-` or `->`. In a `PAG`
```
Comment (andrew-causalens) [inline docs/introduction.md:78]: MAG / PAG, not just PAG.

File: docs/introduction.md [hunk 4]
```diff
@@ -33,35 +33,48 @@ class.
...
+the `--` that is a possible outcome of a wildcard edge (for example `o-`) can also resolve to no edge et all.
```
Comment (andrew-causalens) [inline docs/introduction.md:79]: Not sure about this. How is it no edge? That would maybe happen in the `DAGSampler` but not in this package.

**Review by andrew-causalens — APPROVED** [review]
Thanks for this @max-causalens. Way better than prior docs. I will push one commit to correct a few items then merge and release. Thanks again.

**Pull request merged**